tools/sensors/providers/dwd.py

```diff
@@ -99,33 +98,33 @@ async def fetch_job(self, timestamp: int):
         except BaseException as e:
             logging.error(f"Error fetching DWD data for timestamp {timestamp}: {e}")
 
-    async def _get_meta_file_checksum(self) -> str | None:
+    async def _get_directory_checksum(self) -> str | None:
         """
-        Download the meta-file and calculate its checksum to detect changes.
+        Download the directory listing and calculate its checksum to detect changes.
 
         Returns
         -------
         str | None
-            The SHA-256 checksum of the meta-file or None if failed
+            The SHA-256 checksum of the directory listing HTML or None if failed
         """
         try:
             async with aiohttp.ClientSession(timeout=aiohttp.ClientTimeout(total=self._timeout)) as session:
-                async with session.get(self.META_FILE_URL) as response:
+                async with session.get(self.BASE_URL) as response:
                     if response.status == 200:
-                        content = await response.read()
-                        checksum = hashlib.sha256(content).hexdigest()
+                        content = await response.text()
+                        checksum = hashlib.sha256(content.encode("utf-8")).hexdigest()
```

Comment:
looks like can be simplified
```suggestion
                        content = await response.read()
                        checksum = hashlib.sha256(content).hexdigest()
```